airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/schemas/balances.json

```diff
@@ -0,0 +1,59 @@
+{
+  "$schema": "http://json-schema.org/schema#",
+  "type": "object",
+  "properties": {
+    "balance": {
+      "type": "object",
+      "properties": {
+        "currency": {
+          "type": "string"
```

Comment:
there are a lot of examples in other source's schemas when type is specified like:

a) "type": "string",           - for required fields only
b) "type": ["null", "string"], - for the rest (optional) fields

1. I had a question about judgment about when we should use a) and when b).
How can we decide which field is really required and which is not? Is there any rules?
Is it better to use b) for all fields by default?

2. Do you have any tools why automate schemas creation? Do you make them manually?

Reply:
1. Ok
2. actually, @bazarnov has already created a small tool for this purpose. so it would be good to have 'approved' tools for schema generation.